rootex/framework/scene.cpp

```diff
@@ -68,7 +73,8 @@ Ptr<Scene> Scene::Create(const JSON::json& sceneData, const bool assignNewIDs)
 	}
 	else
 	{
-		thisSceneID = NextSceneID;
+		thisSceneID = std::max(std::max(BaseID, NextSceneID), (SceneID)4);
+		NextSceneID = std::max(BaseID, (SceneID)4);
```

Comment:
This seems wrong.
Should be replaced by 
`NextSceneID = std::max(std::max(BaseID, NextSceneID), (SceneID)4);`  
`thisSceneID = NextSceneID;`
Also, before using 4 explicitly, define a variable for it, eg. `SceneIDOffset`